### PR TITLE
feat(plugin): Simplified markdown experience

### PR DESCRIPTION
**Pull request by sklinefelter**
Updates:

- Leverages constants
- Adds an option to only report classes with issues in the markdown
- Warns and errors less at the top-level
- Adds/updates tests

### CODE REVIEW

**Review by imminent — COMMENTED** [review]
It appears this is accomplishing 2 important/different features: updating the reporting on PRs and correcting how the plugin is able to detect the coverage of nested classes. Ideally those would be in separate PRs that can call out specifically each feature's addition instead of "Update plugin.rb" (which doesn't convey what changed). Other than that, this seems fine

File: spec/fixtures/Blah.java [hunk 1]
```diff
@@ -0,0 +1 @@
+public class Blah
```
Comment (asmenter360) [inline spec/fixtures/Blah.java:1]: Suggestion:
Can we name this more appropriately or just add comment for it's desired use case?

File: spec/fixtures/BlahBlah.kt [hunk 2]
```diff
@@ -0,0 +1 @@
+class BlahBlah
```
Comment (asmenter360) [inline spec/fixtures/BlahBlah.kt:1]: Suggestion:
Can we name this more appropriately or just add comment for it's desired use case?

**Review by asmenter360 — APPROVED** [review]
No issues from my perspective, although I am no Ruby expert so not sure if my review is worth much here.

File: spec/fixtures/Blah.java (hunk 1 above)
Comment (imminent) [reply spec/fixtures/Blah.java]: looks like TestJavaClass

File: spec/fixtures/BlahBlah.kt (hunk 2 above)
Comment (imminent) [reply spec/fixtures/BlahBlah.kt]: Looks like TestKotlinClass

**Review by imminent — COMMENTED** [review]
👍

**Pull request merged**